core/src/main/java/org/apache/gravitino/storage/relational/mapper/provider/base/JobMetaBaseSQLProvider.java

```diff
@@ -0,0 +1,173 @@
+        + " AND jtm.deleted_at = 0";
+  }
+
+  public String listJobPOsByMetalakeAndTemplate(
+      @Param("metalakeName") String metalakeName,
+      @Param("jobTemplateName") String jobTemplateName) {
+    return "SELECT jrm.job_run_id AS jobRunId, jtm.job_template_name AS jobTemplateName,"
+        + " jrm.metalake_id AS metalakeId, jrm.job_execution_id AS jobExecutionId,"
+        + " jrm.job_run_status AS jobRunStatus, jrm.job_finished_at AS jobFinishedAt, "
+        + " jrm.audit_info AS auditInfo,"
+        + " jrm.current_version AS currentVersion, jrm.last_version AS lastVersion,"
+        + " jrm.deleted_at AS deletedAt"
+        + " FROM "
+        + JobMetaMapper.TABLE_NAME
+        + " jrm JOIN "
+        + JobTemplateMetaMapper.TABLE_NAME
+        + " jtm ON jrm.job_template_id = jtm.job_template_id"
+        + " JOIN "
+        + MetalakeMetaMapper.TABLE_NAME
+        + " mm ON jrm.metalake_id = mm.metalake_id"
+        + " WHERE mm.metalake_name = #{metalakeName} AND jtm.job_template_name = #{jobTemplateName}"
+        + " AND jrm.deleted_at = 0 AND mm.deleted_at = 0 AND jtm.deleted_at = 0";
+  }
+
+  public String selectJobPOByMetalakeAndRunId(
+      @Param("metalakeName") String metalakeName, @Param("jobRunId") Long jobRunId) {
+    return "SELECT jrm.job_run_id AS jobRunId, jtm.job_template_name AS jobTemplateName,"
+        + " jrm.metalake_id AS metalakeId, jrm.job_execution_id AS jobExecutionId,"
+        + " jrm.job_run_status AS jobRunStatus, jrm.job_finished_at AS jobFinishedAt,"
+        + " jrm.audit_info AS auditInfo,"
+        + " jrm.current_version AS currentVersion, jrm.last_version AS lastVersion,"
+        + " jrm.deleted_at AS deletedAt"
+        + " FROM "
+        + JobMetaMapper.TABLE_NAME
+        + " jrm JOIN "
+        + JobTemplateMetaMapper.TABLE_NAME
+        + " jtm ON jrm.job_template_id = jtm.job_template_id"
+        + " JOIN "
+        + MetalakeMetaMapper.TABLE_NAME
+        + " mm ON jrm.metalake_id = mm.metalake_id"
+        + " WHERE mm.metalake_name = #{metalakeName} AND jrm.job_run_id = #{jobRunId}"
+        + " AND jrm.deleted_at = 0 AND mm.deleted_at = 0 AND jtm.deleted_at = 0";
+  }
+
+  public String softDeleteJobMetaByMetalakeAndTemplate(
+      @Param("metalakeName") String metalakeName,
+      @Param("jobTemplateName") String jobTemplateName) {
+    return "UPDATE "
+        + JobMetaMapper.TABLE_NAME
+        + " SET deleted_at = (UNIX_TIMESTAMP() * 1000.0)"
+        + " + EXTRACT(MICROSECOND FROM CURRENT_TIMESTAMP(3)) / 1000.0"
+        + " WHERE metalake_id = ("
+        + " SELECT metalake_id FROM "
+        + MetalakeMetaMapper.TABLE_NAME
+        + " WHERE metalake_name = #{metalakeName} AND deleted_at = 0)"
+        + " AND job_template_id = ("
+        + " SELECT job_template_id FROM "
+        + JobTemplateMetaMapper.TABLE_NAME
+        + " WHERE job_template_name = #{jobTemplateName} AND deleted_at = 0)"
```

Comment:
Since the unique key is `metalake_id, job_template_name, deleted_at `, the result of 
```
SELECT job_template_id FROM "
        + JobTemplateMetaMapper.TABLE_NAME
        + " WHERE job_template_name = #{jobTemplateName} AND deleted_at = 0
```

can be larger than 1, I'm afraid that we should not use `job_template_id =` here.

Reply:
Normally this will not be happend, job template id is a unique id.

Reply:
Assuming there are two templates with the same name under two metalakes, this could be a possible scenario.

Reply:
I see, I will fix it.